File: Java/Data Structures & Algorithms Module - Java/Week10/BSTNode.java
/*
BSTNode.java
Billy Cussen
24/11/2020 - Updated 25/11/2020
*/

public class BSTNode<T>{

    T element;
    BSTNode<T> left;
    BSTNode<T> right;

    public BSTNode(T element){
        this.element = element;
    }

    @Override
    public String toString(){
        String leftNode = this.left == null ? "NULL" : this.left.element.toString();
        String rightNode = this.right == null ? "NULL" : this.right.element.toString();
        return "Current: "+this.element+", Left: "+leftNode+", Right: "+rightNode;
    }
}

File: Python/Data Structures & Algorithms Module - Python/Week10/BST Attempt 4/BSTNode.py
"""
BSTNode.py
Billy Cussen
10/12/2020
"""

class BSTNode:

    def __init__(self, element):
        self.element = element
        self.left = None
        self.right = None


    def toString(self):
        leftNode = "NULL" if self.left == None else str(self.left.element)
        rightNode = "NULL" if self.right == None else str(self.right.element)
        return "Current: "+str(self.element)+", Left: "+leftNode+", Right: "+rightNode
